django/db/models/fields/related.py

```diff
@@ -1030,8 +1036,59 @@ def _check_on_delete(self):
                     id="fields.E321",
                 )
             ]
-        else:
-            return []
+        elif (
+            on_delete in [DB_CASCADE, DB_SET_NULL]
+            and hasattr(self.model, "_meta")
+            and any(
+                not parent._meta.abstract
+                for parent in self.model._meta.get_parent_list()
+            )
+        ):
+            return [
+                checks.Error(
+                    "Field specifies unsupported on_delete=DB_CASCADE, on "
```

Comment:
#This check will trigger for both `DB_CASCADE` and `DB_SET_NULL`, but only one of these is mentioned in the error message. We should mention both or parametrise the error message.

The same applies below.